### PR TITLE
[ Code integration in progress ] Add option version_dedup to Job and Pipeline to  filter redundant versions from export

### PR DESCRIPTION
**Pull request by medamineziraoui**

### CODE REVIEW

**Comment by medamineziraoui** [comment]
## Why ?

We want to add an option `version_dedup` for the for the build file of `job` and `pipeline` that will control the versions exportation for the v1.

## Links / Ressources

https://github.com/saagie/gradle-saagie-dataops-plugin/issues/256

## Actual  status / Investigations

if we set  `include_all_versions=true` `pipeline`/ `job` to true and use `projectsExportV1` it will export the job/pipeline concerned and export all versions found in the v1. but it will **only export different versions** automaticaly, and we want to controll this.

## How?

- [ ] Add property version_dedup with false value to job entity.
- [ ] Add property version_dedup with false value to pipeline entity.
- [ ] Change condition for removing duplicate versions, if `version_dedup =true` then remove duplicate. if false then keep all versions for job.
- [ ] Change condition for removing duplicate versions, if `version_dedup =true` then remove duplicate. if false then keep all versions for pipeline.
- [ ] Fix the bug that happens when removing current remove duplication logic. ( one of the version is missing )

File: src/main/groovy/io/saagie/plugin/dataops/models/Job.groovy [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package io.saagie.plugin.dataops.models
 
+import com.sun.org.apache.xpath.internal.operations.Bool
```
Comment (DecampsRenan) [inline src/main/groovy/io/saagie/plugin/dataops/models/Job.groovy:3]: Is this import used ?

File: src/main/groovy/io/saagie/plugin/dataops/utils/SaagieUtils.groovy [hunk 2]
```diff
@@ -1249,6 +1249,25 @@ class SaagieUtils {
 		outputStream.close()
 		inputStream.close()
 	}
+	// This function is helpful to get for example size of a file without downloading it
+	
+	int getOnlyHeaderInformationFromApi(String url, OkHttpClient client, version) {
```
Comment (DecampsRenan) [inline src/main/groovy/io/saagie/plugin/dataops/utils/SaagieUtils.groovy:1254]: If it's only for file, maybe rename it to getFileInformationsFromApi, I think it's enougth

File: src/main/groovy/io/saagie/plugin/dataops/utils/SaagieUtils.groovy [hunk 3]
```diff
@@ -1249,6 +1249,25 @@ class SaagieUtils {
...
+			logger.debug("getting file size from header for v1 ${version}....")
+			Request request = this.buildRequestForFileHeader(url)
+			response = client.newCall(request).execute()
+			return response.body().contentLength();
+		} catch (IOException e) {
+			if (response!=null) {
+				response.close();
+				
+			}
+			e.printStackTrace();
```
Comment (DecampsRenan) [inline src/main/groovy/io/saagie/plugin/dataops/utils/SaagieUtils.groovy:1266]: Should we keep that ? I think the error should go to an error or debug statement

**Pull request closed without merging**